drivers/analog-io/max2201x/iio_max2201x.c

```diff
@@ -0,0 +1,919 @@
+		.show = max2201x_iio_read_config_available,
+		.shared = IIO_SHARED_BY_ALL
+	},
+	{
+		.name = "overcurrent_shutdown",
+		.store = max2201x_iio_write_config,
+		.priv = MAX2201X_IIO_OVC_SHTDN,
+	},
+	{
+		.name = "overcurrent_shutdown_available",
+		.show = max2201x_iio_read_config_available,
+		.shared = IIO_SHARED_BY_ALL
+	},
+	{
+		.name = "currrent_limit",
+		.store = max2201x_iio_write_config,
+		.priv = MAX2201X_IIO_CURR_LIM,
+	},
+	{
+		.name = "current_limit_available",
+		.show = max2201x_iio_read_config_available,
+		.shared = IIO_SHARED_BY_ALL
+	},
+	{
+		.name = "dac_ref",
+		.store = max2201x_iio_write_config,
+		.priv = MAX2201X_IIO_DAC_REF,
+	},
+	{
+		.name = "dac_ref_available",
+		.show = max2201x_iio_read_enable_available,
+		.shared = IIO_SHARED_BY_ALL
+	},
+	{
+		.name = "interrupts",
+		.show = max2201x_iio_read_interrupt
+	},
+	{
+		.name = "interrupts_enable",
+		.store = max2201x_iio_write_int_en,
+	},
+	{
+		.name = "ld_ctrl",
+		.store = max2201x_iio_write_config,
+		.priv = MAX2201X_IIO_LD_CTRL,
+	},
+	{
+		.name = "ld_ctrl_available",
+		.show = max2201x_iio_read_enable_available,
+		.shared = IIO_SHARED_BY_ALL
+	},
+	{
+		.name = "gain",
+		.store = max2201x_iio_write_gain,
+	},
+	{
+		.name = "slew_rate",
+		.store = max2201x_iio_write_slew_rate,
+		.priv = MAX2201X_IIO_SLEW_RATE,
+	},
+	{
+		.name = "slew_rate_available",
+		.show = max2201x_iio_read_enable_available,
+	},
+	{
+		.name = "range",
+		.store = max2201x_iio_write_slew_rate,
+		.priv = MAX2201X_IIO_RANGE,
+	},
+	{
+		.name = "range_available",
+		.show = max2201x_iio_read_range_available,
+	},
+	{
+		.name = "step_size",
+		.store = max2201x_iio_write_slew_rate,
+		.priv = MAX2201X_IIO_STEP_SIZE,
+	},
+	{
+		.name = "step_size_available",
+		.show = max2201x_iio_read_step_size_available,
+		.shared = IIO_SHARED_BY_ALL,
+	},
+	{
+		.name = "update_rate",
+		.store = max2201x_iio_write_slew_rate,
+		.priv = MAX2201X_IIO_UPDATE_RATE,
+	},
+	{
+		.name = "update_rate_available",
+		.show = max2201x_iio_read_update_rate_available,
+		.shared = IIO_SHARED_BY_ALL,
+	},
+	{
+		.name = "operation_mode",
+		.store = max2201x_iio_write_op_mode,
+	},
+	{
+		.name = "operation_mode_available",
+		.show = max2201x_iio_read_op_mode_available,
+		.shared = IIO_SHARED_BY_ALL,
+	},
+	{
+		.name = "polarity",
+		.store = max2201x_iio_write_ao_config,
+		.priv = MAX2201X_IIO_POLARITY,
+	},
+	{
+		.name = "polarity_available",
+		.show = max2201x_iio_read_polarity_available,
+		.shared = IIO_SHARED_BY_ALL,
+	},
+	{
+		.name = "cm_sense",
+		.store = max2201x_iio_write_ao_config,
+		.priv = MAX2201X_IIO_CM_SENSE,
+	},
+	{
+		.name = "cm_sense_available",
+		.show = max2201x_iio_read_cm_sense_available,
+		.shared = IIO_SHARED_BY_ALL,
+	},
+	{
+		.name = "ldac",
+		.store = max2201x_iio_write_ao_config,
+		.priv = MAX2201X_IIO_LDAC,
+	},
+	{
+		.name = "ldac_available",
+		.show = max2201x_iio_read_enable_available,
+		.shared = IIO_SHARED_BY_ALL
```

Comment:
the shared field is not relevant for a debug attribute (only relevant for channel attributes, should be removed everywhere where applicable)